test/e2e/app/snapshots.go

```diff
@@ -16,6 +16,9 @@ import (
 
 const (
 	snapshotChunkSize = 1e6
+
+	// keep only the most recent 10 snapshots. Older snapshots are pruned
```

Comment:
```suggestion
	// Keep only the most recent 10 snapshots. Older snapshots are pruned
```

Reply:
why 10?

Reply:
Just an estimated, yet arbitrary number. I think the minimum snapshot interval is 3. So that would make 30 blocks to statesync. I figured this would be enough, however the e2e tests have proven me wrong on many of an occasion. I'm happy to change it if you have a more reasonable suggestion. 

As a side note, It would be interesting to test removing a snapshot halfway through a node that is statesyncing on that snapshot

Reply:
> As a side note, It would be interesting to test removing a snapshot halfway through a node that is statesyncing on that snapshot

It's like you've _seen_ my backup directory. 😉

Reply:
> Just an estimated, yet arbitrary number. I think the minimum snapshot interval is 3. So that would make 30 blocks to statesync. I figured this would be enough, however the e2e tests have proven me wrong on many of an occasion. I'm happy to change it if you have a more reasonable suggestion.

I'd make it be some multiplier on top of a basic minimum? (or like numNodes*10)  If you have bigger nodes (and we relax the cap of 2 statesync per generated network) and run for longer, then I could see it becoming an issue, not sure. Having said that I'm sure it's fine for now

> As a side note, It would be interesting to test removing a snapshot halfway through a node that is statesyncing on that snapshot

"interesting" as in "this should work fine?" or "interesting" as in "it'd be cool to see how it crashes?"

Reply:
> "interesting" as in "this should work fine?" or "interesting" as in "it'd be cool to see how it crashes?"

I would like to think it works - it just gives up trying to fetch the remaining chunks and tries again with a more recent snapshot

Reply:
> I'd make it be some multiplier on top of a basic minimum?

I don't think variances in the testnet structure will make it more or less likely that a node prunes the snapshot that it is also providing to a state syncing node. The only thing that would make sense is to make it a multiplier of the snapshot interval. A small snapshot interval should equate to having more snapshots stored at any given moment. A large snapshot interval should equate to less snapshots stored at a given moment. We could do something like:
```
maxAmountOfSnapshots = 5 + 30/snapshotInterval
```
So for e2e testnets with a snapshot interval of 3 that is 15 snapshots. For a long lived network with a snapshot interval of 100 that is 6 snapshots. I still think this is probably a bit overkill and having a fixed number should suffice

Reply:
I like using a multiplier/factor of snapshot interval, but yeah, I think I've come around on this.